src/common/valgen.cpp

```diff
@@ -172,6 +172,28 @@ bool wxGenericValidator::TransferToWindow()
             pControl->SetValue(*m_pBool) ;
             return true;
         }
+        else if (m_pInt)
+        {
+            wxRadioButton* selected = nullptr;
+            for (int i = 0 ;
+                pControl ;
+                ++i, pControl = pControl != pControl->GetLastInGroup() ? pControl->GetNextInGroup() : nullptr)
```

Comment:
Sorry, I find this loop pretty difficult to read. I see a few different ways to simplify it, but I'd probably go with

```suggestion
            for ( int i = 0; ; ++i )
```

here and then add

```c++
if ( pControl == pControl->GetLastInGroup() )
    break;

pControl = pControl->GetNextInGroup()
```

at the end of the loop.